Dockerfile

```diff
@@ -56,6 +56,14 @@ WORKDIR /opt/guacamole
 # Copy artifacts from builder image into this image
 COPY --from=builder /opt/guacamole/ .
 
+# Turn on the Remote IP Valve
+RUN sed -i 's+^\(      </Host>\)+        <Valve className="org.apache.catalina.valves.RemoteIpValve" />\n\1+' /usr/local/tomcat/conf/server.xml
+
+# Allow image to run as non-root user
+RUN sed -i 's/unpackWARs="true"/unpackWARs="false"/' /usr/local/tomcat/conf/server.xml
+RUN chmod 777 /usr/local/tomcat/temp
```

Comment:
This looks dangerous. Reducing privileges is a good thing, but removing existing restrictions from Tomcat's internals in order to achieve that seems a step in the wrong direction. That would be effectively granting privileges.

Assuming it is possible to configure the Tomcat portion of the image such that it can be run as a lower-privilege user, we should find a way to do that which doesn't lessen the security restrictions already in place.

Reply:
Thank you @mike-jumper for your great review.
I think you talk here about the temp directory permissions.
I though rather strange that, by default, `temp` directory was not world-writable.
For sure, being non-writable by non-root user prevents the non-unpacked app to start.
We could workaround this by setting the runtime `CATALINA_TMPDIR` env var to another writable dir, `/tmp` for instance. Let me then know if you think we could get rid of this chmod command.

Reply:
There's definitely a difference in purpose between `/tmp` (the system-wide temp directory) and `CATALINA_TMPDIR` (the temp directory dedicated to Tomcat). The former needs to be world-writable, yes, but I don't know what the security implications would be of making the Tomcat-specific temp directory world-writable.

The fact that:

* A system-wide, generic temp directory isn't already used for `CATALINA_TMPDIR`
* The Tomcat-specific directory isn't world-writable by default, but rather locked down for Tomcat's use.

indicates to me that this change should be made with caution. I think we would need to firmly establish that this directory is safe to be world-writable.

One alternative could be to create a Tomcat-specific subdirectory within `/tmp` or `/var/tmp` and point `CATALINA_TMPDIR` to that. As long as that subdirectory has the same permissions as the default `CATALINA_TMPDIR` normally would, this should be safe. Pointing `CATALINA_TMPDIR` directly to `/tmp` itself would not necessarily be safe unless it can be established that what you're already doing (`chmod 777`) is safe.

Reply:
I agree.
I then just pushed a much better way 👍